bdk_chain/src/spk_txout_index.rs

```diff
@@ -64,18 +64,28 @@ impl<I: Clone + Ord> SpkTxOutIndex<I> {
     /// See [`ForEachTxout`] for the types that support this.
     ///
     /// [`ForEachTxout`]: crate::ForEachTxout
-    pub fn scan(&mut self, txouts: &impl ForEachTxout) {
-        txouts.for_each_txout(&mut |(op, txout)| self.scan_txout(op, txout))
+    pub fn scan(&mut self, txouts: &impl ForEachTxout) -> BTreeSet<I> {
```

Comment:
`BTreeSet<&I>` to be consistent with rest of API.

Reply:
@LLFourn I tried, but lifetime issues because we can't take reference out of closure apparently?

Reply:
I also can't fix this. Maybe remove the return value here?

Reply:
Yeah this is good just make a small internal function that is called by both functions?